flare_portal/experiments/views.py

```diff
@@ -522,17 +524,33 @@ def get_context_data(self, **kwargs: Any) -> dict:
 class ParticipantFormSetView(FormView):
     form_class = ParticipantFormSet  # type: ignore
     template_name = "experiments/participant_list.html"
+    paginate_by = settings.DEFAULT_PER_PAGE
 
     def dispatch(self, *args: Any, **kwargs: Any) -> HttpResponse:
         self.experiment = get_object_or_404(Experiment, pk=kwargs["experiment_pk"])
         return super().dispatch(*args, **kwargs)
 
     def get_form_kwargs(self) -> dict:
+        self.page = self.request.GET.get("page", 1)
+        participants_qs = Participant.objects.filter(
+            experiment=self.experiment,
+        )
+        paginator = Paginator(participants_qs, self.paginate_by)
+        try:
+            participants = paginator.page(self.page)
+        except PageNotAnInteger:
+            participants = paginator.page(1)
+        except EmptyPage:
+            participants = paginator.page(paginator.num_pages)
```

Comment:
**suggestion:** Handle `InvalidPage` here, to ensure requests for pages out of range don't cause an error.